src/Projection/PdoEventStoreProjection.php

```diff
@@ -725,10 +693,67 @@ private function persist(): void
 
         $statement = $this->connection->prepare($sql);
         $statement->execute([
-            json_encode($this->streamPositions),
-            json_encode($this->state),
+            json_encode($this->streamPositions, \JSON_FORCE_OBJECT),
+            json_encode($this->state, \JSON_FORCE_OBJECT),
             $lockUntilString,
             $this->name,
         ]);
     }
+
+    private function prepareStreamPositions(): void
+    {
+        $streamPositions = [];
+
+        if (isset($this->query['all'])) {
+            $sql = <<<EOT
+SELECT real_stream_name FROM $this->eventStreamsTable WHERE real_stream_name NOT LIKE '$%';
```

Comment:
Maybe we should stay consistent and always add backticks around field names in sql queries.